### PR TITLE
Per tenant max search duration

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
Adds the ability to have a per tenant max search duration. By default, if there is no per tenant configuration, the original config value is used. This PR also updates the jsonnet under `./operations` to mount the overrides yaml onto the frontend.

Additional changes
- Documentation as well as catching the missing `block_retention` override
- Updated the jsonnet example to use full backend search

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/tempo/website/configuration/_index.md [hunk 1]
```diff
@@ -987,7 +987,15 @@ overrides:
...
     # actually writing these metrics.
     [metrics_generator_disable_collection: <bool> | default = false]
-      
+
+    # Per-user block retention. If this value is set to 0 (default) then block_retention
+    #  in the compactor config is used.
+    [block_retention: <duration> | default = 0s]
+
+    # Per-user max search duration. If this value is set to 0 (default) then max_duration
+    #  in the frontend config is used.
```
Comment (knylander-grafana) [inline docs/tempo/website/configuration/_index.md:996]:
```suggestion
    # Per-user block retention. If this value is set to 0 (default), then block_retention
    #  in the compactor configuration is used.
    [block_retention: <duration> | default = 0s]

    # Per-user max search duration. If this value is set to 0 (default), then max_duration
    #  in the front-end configuration is used.
```

**Review by knylander-grafana — CHANGES_REQUESTED** [review]
Reviewed the doc files and have a few minor suggestions. "Config" was changed to configuration so it's spelled out. Front-end component is a compound adjective so should be hyphenated.

**Review by knylander-grafana — APPROVED** [review]
Thank you for making the suggested changes. I'll use Comment instead next time.

File: modules/overrides/limits.go [hunk 2]
```diff
@@ -70,6 +70,9 @@ type Limits struct {
 	// Querier enforced limits.
 	MaxBytesPerTagValuesQuery int `yaml:"max_bytes_per_tag_values_query" json:"max_bytes_per_tag_values_query"`
 
+	// QueryFrontend enforced limits
+	MaxSearchDuration model.Duration `yaml:"max_search_duration" json:"max_search_duration"`
```
Comment (mdisibio) [inline modules/overrides/limits.go:74]: Is there a reason for `model.Duration`?  Just comparing to `MetricsGeneratorCollectionInterval time.Duration` above.

**Review by mdisibio — APPROVED** [review]
LGTM just one question.

**Pull request merged**